backend/app/utils/admin_logger.py

```
@@ -0,0 +1,243 @@
+async def ensure_admin_logs_table() -> bool:
+    """Check whether the admin_logs table is available in the current environment."""
+    global _admin_logs_table_ready
+
+    if _admin_logs_table_ready is True:
+        return True
+
+    try:
+        supabase = get_supabase_client()
+        await supabase.table("admin_logs").select("id").limit(1).execute()
+        _admin_logs_table_ready = True
+        return True
```

Ruff (0.15.0)

[warning] Consider moving this statement to an `else` block

(TRY300)